includes/admin/class-sensei-admin-notices.php

```diff
@@ -521,6 +535,40 @@ private function condition_installed_since( $installed_since ) : bool {
 		return $installed_at <= $installed_since;
 	}
 
+	/**
+	 * Check a date range condition.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param ?string $start_date_str Start date.
+	 * @param ?string $end_date_str   End date.
+	 *
+	 * @return bool
+	 */
+	private function condition_check_date_range( ?string $start_date_str, ?string $end_date_str ) : bool {
+		$condition_pass = true;
+		$now            = new DateTime();
+
+		// Defaults to WP timezone, but can be overridden by passing string that includes timezone.
+		$start_date = $start_date_str ? date_create( $start_date_str, wp_timezone() ) : null;
+		$end_date   = $end_date_str ? date_create( $end_date_str, wp_timezone() ) : null;
+
+		// If the passed date strings are invalid, don't show the notice.
+		if ( false === $start_date || false === $end_date ) {
+			return false;
+		}
+
+		if ( $start_date && $now < $start_date ) {
+			$condition_pass = false;
```

Comment:
Not critical, but maybe the 3 conditions could `return false` directly.

Reply:
Good idea! Changed in b362f2f52e8085c5ffc162b52a989428026fcc5b